pkg/policy/repository.go

```diff
@@ -122,43 +122,53 @@ func (p *Repository) AllowsIngressLabelAccess(ctx *SearchContext) api.Decision {
 	return decision
 }
 
+// wildcardL3L4Rule is a post-processing stage for the L4PolicyMap which scans
+// through the map for filters whose L7 rules would be shadowed by an L3/L4
+// rule (specified as 'proto', 'port', 'endpoints'), and ensures that the
+// filter is configured such that all L7 traffic is then allowed on the port.
+//
+// If there are other L7 rules for the same L3 peer on a port, then the L7
+// rules are set to allow all; if there are no L7 rules for the same L3 peer
+// then the 'L4Filter.L7RulesPerEp' left unconfigured for the L3 peer L7 rules.
+//
+// This function is co-dependent on mergeL4Port()'s handling of rules that use
+// the wildcard selector; it relies on mergeL4Port inserting shadowed L3
+// selectors into the 'L4Filter.Endpoints' slice.
```

Comment:
A++ documentation! Great job - stuff like this is so important for code maintainability.